src/checkpoint.hh

```diff
@@ -0,0 +1,140 @@
+/* {{{
+
+    This file is part of gapc (GAPC - Grammars, Algebras, Products - Compiler;
+      a system to compile algebraic dynamic programming programs)
+
+    Copyright (C) 2008-2011  Georg Sauthoff
+         email: gsauthof@techfak.uni-bielefeld.de or gsauthof@sdf.lonestar.org
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+}}} */
+
+#ifndef SRC_CHECKPOINT_HH_
+#define SRC_CHECKPOINT_HH_
+
+#include <string>
+#include "printer.hh"
+#include "symbol.hh"
+#include "statement/table_decl.hh"
+
+namespace Printer {
+class Checkpoint : public Base {
+ public:
+    size_t interval = 0;
+
+    explicit Checkpoint(size_t interval): interval(interval) {}
+
+    void include(Printer::Base &stream) {
+      stream << "#include \"boost/serialization/vector.hpp\"" << endl;
```

Comment:
I feel not all dependencies are currently defined in the configuration of the compiler:
```
(base) t490s x86_64 /Daten/Git/jlab/gapc>make -f out_cp.mf
Including global makefile /home/sjanssen/share/gapc/config_linux-gnu.mf
echo '#include "out_cp.hh"' > out_cp_main.cc
cat /home/sjanssen/include/rtlib/generic_main.cc >> out_cp_main.cc
g++  -I/usr/include  -I/usr/include -I/home/sjanssen/include/ -I/home/sjanssen/include/rtlib -I/home/sjanssen/include/librna  -O3  -std=c++17 -D_XOPEN_SOURCE=500 -MMD -MP -Wall -Wnon-virtual-dtor -Wno-unused-variable -Wno-parentheses -DNDEBUG -DBISONNEW   out_cp_main.cc -c -o out_cp_main.o 
g++  -I/usr/include  -I/usr/include -I/home/sjanssen/include/ -I/home/sjanssen/include/rtlib -I/home/sjanssen/include/librna  -O3  -std=c++17 -D_XOPEN_SOURCE=500 -MMD -MP -Wall -Wnon-virtual-dtor -Wno-unused-variable -Wno-parentheses -DNDEBUG -DBISONNEW   out_cp.cc -c -o out_cp.o 
g++  -I/usr/include  -I/usr/include -I/home/sjanssen/include/ -I/home/sjanssen/include/rtlib -I/home/sjanssen/include/librna  -O3  -std=c++17 -D_XOPEN_SOURCE=500 -MMD -MP -Wall -Wnon-virtual-dtor -Wno-unused-variable -Wno-parentheses -DNDEBUG -DBISONNEW   -c /home/sjanssen/include/rtlib/string.cc -o string.o
g++  -o out_cp out_cp_main.o out_cp.o string.o  -L/usr/lib/x86_64-linux-gnu  -L/home/sjanssen/lib/ -L/home/sjanssen/lib/rtlib -L/home/sjanssen/lib/librna   -Xlinker -rpath -Xlinker /usr/lib/x86_64-linux-gnu      -Xlinker -rpath -Xlinker /home/sjanssen/lib/ -Xlinker -rpath -Xlinker /home/sjanssen/lib/rtlib -Xlinker -rpath -Xlinker /home/sjanssen/lib/librna    -L/usr/lib/x86_64-linux-gnu -lgsl -lgslcblas -lm -lrna  -lboost_serialization
/usr/bin/ld: out_cp_main.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make: *** [out_cp.mf:35: out_cp] Error 1
```

Reply:
works with call `make -f out_cp.mf LDFLAGS_EXTRA=" -lpthread "`

Reply:
Interesting, I didn't need anything other than `-lboost_serialization` for the compilation/linking to work. I will add `-lpthread` to the Makefile printer then.